trie/verkle.go

```diff
@@ -104,12 +125,23 @@ func (t *VerkleTrie) TryGetAccount(key []byte) (*types.StateAccount, error) {
 	if len(values[utils.NonceLeafKey]) > 0 {
 		acc.Nonce = binary.LittleEndian.Uint64(values[utils.NonceLeafKey])
 	}
-	var balance [32]byte
-	if len(values[utils.BalanceLeafKey]) > 0 {
-		for i := 0; i < len(balance); i++ {
-			balance[len(balance)-i-1] = values[utils.BalanceLeafKey][i]
-		}
+	// if the account has been deleted, then values[10] will be 0 and not nil. If it has
+	// been recreated after that, then its code keccak will NOT be 0. So return `nil` if
+	// the nonce, and values[10], and code keccak is 0.
+	if acc.Nonce == 0 && len(values) > 10 && len(values[10]) > 0 && bytes.Equal(values[utils.CodeKeccakLeafKey], zero[:]) {
+		return nil, errDeletedAccount
 	}
+	var balance [32]byte
+	copy(balance[:], values[utils.BalanceLeafKey])
+	for i := 0; i < len(balance)/2; i++ {
+		balance[len(balance)-i-1], balance[i] = balance[i], balance[len(balance)-i-1]
+	}
+	// var balance [32]byte
+	// if len(values[utils.BalanceLeafKey]) > 0 {
+	// 	for i := 0; i < len(balance); i++ {
+	// 		balance[len(balance)-i-1] = values[utils.BalanceLeafKey][i]
+	// 	}
+	// }
```

Comment:
@jsign this is the result of a merge conflict. I think the one that is commented is the one that you introduced (I could be wrong) for performance purposes. It also seems to return the wrong commitments in the verkle processing test (well, at least _different_ commitments from what that tests display in beverly hills). I assume that `values` are reused. Didn't get to the bottom of this yet, but if you have some input on this, I'm interested.

Reply:
That commented code is a change you did a while ago: https://github.com/gballet/go-ethereum/pull/217
`values` being reused shouldn't be a problem in both since they're copying bytes.

From my perspective, both the commented and uncommented should work fine.

(Yes, the commented one is doing less work since the uncommented is copying the whole slice and doing the swap. The commented one is copying bytes directly in the right order. But honestly, this perf difference should be negligible in the gran scheme of things)